src/Dialect/Mlir/DialectBuilder.cpp

```diff
@@ -635,8 +635,9 @@ Value MathBuilder::cast(Type destType, Value src) const {
     // TosaToLinalg in MLIR uses a fancier algorithm that clamps values to
     // min/max signed/unsigned integer values.
     if (destType.isUnsignedInteger()) {
-      Value cast = castToSignless(src, srcWidth);
-      return b().create<arith::FPToUIOp>(loc(), destType, cast);
+      Type castType = b().getIntegerType(destWidth);
+      Value cast = b().create<arith::FPToUIOp>(loc(), castType, src);
+      return castToUnsigned(cast, destWidth);
```

Comment:
Fix an issue when casting from float to unsigned int. In this case, `src` is float, so we cannot call `castToSignless` on `src`